### ISSUE TITLE
Auto restart or reload on plugin installation

### ISSUE THREAD

**Issue by jeremyhart** [QL-Win/QuickLook#406]
The plugin installation process needs to be as seamless as possible, the qlplugin package is a good start, seems unnecessary to manually restart QuickLook, this should be able to be done programmatically. 

QuickLooks future probably relies on plugins so we should put a lot of effort into making it seamless for the user and as well as the devs.

**Comment by xupefei** [comment]
I agree that the current plugin installer is quite shabby and requires some user-friendliness enhancements.
Regarding the auto restart after installing a plugin, what we need is a standalone `runner.exe` which can run `QuickLook.exe` again after we have killed the old process. The exe should belong to PluginInstaller plugin.

Another issue is the uninstallation. I currently have no idea about a good way to allow users to uninstall a pluginwithout adding more windows, e.g., a plugin list window, to QuickLook.

**Comment by jeremyhart** [comment]
Is reloading the plugins without a restart an option?
Until there is a plugin management window, we could just have a link from the tray icon to the plugin folder, users can just delete the folder manually.

**Comment by xupefei** [comment]
> Is reloading the plugins without a restart an option?

No, it's not possible: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/assemblies-gac/how-to-load-and-unload-assemblies

> we could just have a link from the tray icon to the plugin folder

Good idea.

**Comment by Geeyun-JY3** [comment]
What about adding a `restart` button to the context menu of notification(tray) icon? I know it cannot be the solution to this issue. This idea just comes out from my brain and I think QuickLook can implement it first.

**Comment by emako** [comment]
Just add a tray menu button to restart Quicklook #1448
